run-with-macadam.sh

```diff
@@ -0,0 +1,327 @@
+  content: '$PUB_KEY'
+  owner: core
+  permissions: '0600'
+- path: /opt/crc/id_rsa.pub
+  content: '$PUB_KEY'
+  owner: root:root
+  permissions: '0644'
+- path: /etc/sysconfig/crc-env
+  content: |
+    CRC_SELF_SUFFICIENT=1
+    CRC_NETWORK_MODE_USER=1
+  owner: root:root
+  permissions: '0644'
+- path: /opt/crc/pull-secret
+  content: |
+    $PULL_SECRET
+  permissions: '0644'
+- path: /opt/crc/pass_kubeadmin
+  content: '$PASS_KUBEADMIN'
+  permissions: '0644'
+- path: /opt/crc/pass_developer
+  content: '$PASS_DEVELOPER'
+  permissions: '0644'
+- path: /opt/crc/ocp-custom-domain.service.done
+  permissions: '0644'
+EOF
+    USER_DATA_PATH="$(pwd)/user-data"
+    echo "cloud-init user-data file created."
+}
+
+# Extracts the VM disk image from the CRC bundle
+function extract_disk_img() {
+    echo "--- Extracting VM image from CRC bundle ---"
+    if [[ ! -f "${CRC_BUNDLE_PATH}" ]]; then
+        die "CRC bundle not found at ${CRC_BUNDLE_PATH}."
+    fi
+
+    local bundle_name
+    bundle_name=$(basename "${CRC_BUNDLE_PATH}")
+    local disk_image_name
+    disk_image_name="crc.qcow2"
+    tar --zstd -O -xvf "${CRC_BUNDLE_PATH}" "${bundle_name%.*}"/${disk_image_name} >"${disk_image_name}"
+    if [[ ! -s "${disk_image_name}" ]]; then
+        die "Failed to extract disk image from bundle."
+    fi
+    DISK_IMAGE_PATH="$(pwd)/${disk_image_name}"
+    echo "VM image extracted to ${DISK_IMAGE_PATH}."
+}
+
+# Ensures macadam tool is available
+function ensure_macadam_exists() {
+    echo "--- Setting up macadam ---"
+    if command -v macadam &>/dev/null; then
+        echo "macadam is already available in PATH."
+        return
+    fi
+
+    if [ -f "/opt/macadam/macadam" ]; then
+        export PATH="/opt/macadam:$PATH"
+        echo "macadam found in /opt/macadam and added to PATH."
+        return
+    fi
+
+    echo "macadam not found, downloading..."
+
+    local version="latest"
+    local url="https://github.com/crc-org/macadam/releases/download/${version}/macadam-linux-amd64"
+
+    echo "Downloading from ${url}"
+    sudo mkdir -p /opt/macadam
+    if ! curl -fL -o /tmp/macadam "${url}"; then
+        die "Failed to download macadam."
+    fi
+    sudo mv /tmp/macadam /opt/macadam/macadam
+    sudo chmod +x /opt/macadam/macadam
+
+    export PATH="/opt/macadam:$PATH"
+    echo "macadam downloaded to /opt/macadam and added to PATH."
+}
+
+# start the VM
+function start_macadam_vm() {
+    echo "--- Creating VM ---"
+    macadam init \
+        "${DISK_IMAGE_PATH}" \
+        --disk-size 31 \
+        --memory 11264 \
+        --name crc-ng \
+        --username core \
+        --ssh-identity-path "${PRIV_KEY_PATH}" \
+        --cpus 6 \
+        --cloud-init "${USER_DATA_PATH}" \
+        --log-level debug
+
+    if ! macadam start crc-ng --log-level debug; then
+        echo "Machine didn't come up in time"
+    fi
+}
+
+# Exposes a port from the VM to the host
+function forward_port_gvproxy() {
```

Comment:
- need to ensure gvproxy is available on the machine
- run `sudo setcap 'cap_net_bind_service=+ep' gvproxy`
- also forward ports 80 and 443